api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupAnalyticsMetricReport.java

```diff
@@ -0,0 +1,217 @@
+        CohortDefinition enrolledInTheQuarter = hivCohortDefinitionLibrary.getEnrolledInCareBetweenDates();
+
+        CohortDefinition everEnrolledByEndQuarter = df.getPatientsNotIn(enrolledOnOrBeforeQuarter, enrolledInTheQuarter);
+        CohortDefinition enrolledDuringTheQuarter = df.getPatientsNotIn(enrolledInTheQuarter, transferredInTheQuarter);
+
+        CohortDefinition cumulativeEverEnrolled = df.getPatientsInAny(everEnrolledByEndQuarter, enrolledDuringTheQuarter);
+
+        CohortDefinition patientsThroughReception = df.getAnyEncounterOfTypesByEndOfDate(null, Arrays.asList(reception));
+        CohortDefinition patientsThroughTriage = df.getAnyEncounterOfTypesByEndOfDate(null,Arrays.asList(triage));
+        CohortDefinition patientsThroughCounselor = df.getAnyEncounterOfTypesByEndOfDate(null,Arrays.asList(counselor));
+        CohortDefinition patientsThroughLab = df.getAnyEncounterOfTypesByEndOfDate(null,Arrays.asList(Lab));
+        CohortDefinition patientsThroughPharmacy = df.getAnyEncounterOfTypesByEndOfDate(null,Arrays.asList(Pharmacy));
+        CohortDefinition patientsThroughARTClinician = df.getAnyEncounterOfTypesByEndOfDate(null,Arrays.asList(ART_Clinician));
+
+        CohortDefinition transferredOutPatients = hivCohortDefinitionLibrary.getPatientsTransferredOutDuringPeriod();
+        CohortDefinition deadPatientsByEndDate = df.getPatientsInAll(everEnrolledByEndQuarter,df.getDeadPatientsByEndDate());
+        CohortDefinition lost_to_followup = df.getLostToFollowUp();
+
+        CohortDefinition hadEncounterInQuarter = hivCohortDefinitionLibrary.getArtPatientsWithEncounterOrSummaryPagesBetweenDates();
+        CohortDefinition longRefillPatients = df.getPatientsWithLongRefills();
+
+        CohortDefinition patientsWithReturnVisitDuringPeriod = df.getPatientsWhoseObsValueDateIsBetweenStartDateAndEndDate(hivMetadata.getReturnVisitDate(),
+                Arrays.asList(hivMetadata.getARTEncounterEncounterType()), BaseObsCohortDefinition.TimeModifier.ANY);
+
+        CohortDefinition activePatients= df.getPatientsInAny(hadEncounterInQuarter,longRefillPatients,patientsWithReturnVisitDuringPeriod);
+        CohortDefinition activePatientsInCareDuringPeriod = df.getPatientsNotIn(activePatients,df.getPatientsInAny(transferredOutPatients,deadPatientsByEndDate,lost_to_followup));
+
+        CohortDefinition nonSuppressedClientsByEndDate = df.getPatientsWithNumericObsByEndOfPreviousDate(hivMetadata.getViralLoadCopies(),hivMetadata.getARTEncounterPageEncounterType(), RangeComparator.GREATER_EQUAL,1000.0, BaseObsCohortDefinition.TimeModifier.LAST);
+        CohortDefinition suppressedClientsByEndDate = df. getPatientsWithNumericObsByEndOfPreviousDate(hivMetadata.getViralLoadCopies(),hivMetadata.getARTEncounterPageEncounterType(), RangeComparator.LESS_THAN,1000.0, BaseObsCohortDefinition.TimeModifier.LAST);
+
+        addIndicator(dsd, "a", "ART Encounter", ARTEncounter, "");
```

Comment:
@dbaluku Can you give the metrics more descriptive names so that they are easier to troubleshoot later 

a - art_encounter
b - hts_encounter
c - eid_encounter
d - active_patinets_in_care
e - patients_ever_enrolled
f - patients_served_at_reception 

etc